tools/src/commands/GenerateDocsAPIData.ts

```diff
@@ -66,9 +71,10 @@ const executeCommand = async (
     }
 
     if (MINIFY_JSON) {
-      const minifiedJson = recursiveOmitBy(
-        output,
-        ({ key }) => key === 'id' || key === 'groups'
+      const minifiedJson = recursiveOmitBy(output, ({key, node}) =>
+        key === 'id' ||
+        key === 'groups' ||
+        (key === 'flags' && !Object.keys(node).length)
       );
       await fs.writeFile(jsonOutputPath, JSON.stringify(minifiedJson, null, 0));
```

Comment:
Actually I'm wondering if not using `0` here wouldn't be better 🤔 If these jsons are one-liners:
- it's harder to resolve conflicts and spot the differences during review
- git can't optimize changes in these files — git operates on lines, so the delta between subsequent revisions is now the same as the entire content and as a result, it behaves similarly to binary files

They all are not very important for obvious reasons, just expressing my thoughts. For the last one — I might be too obsessive about repo size because it's already too big and cloning it takes a lot of time (I'm working on deleting old objects from the history tho). In that case it doesn't affect its size too much, but just wanted to make you aware of this fact 😉 Optimizing the size of a single file doesn't always give the best optimization 😄 

Anyway, let's leave it as is for now.

Reply:
> Actually I'm wondering if not using `0` here wouldn't be better 🤔

Sorry I didn't get that suggestion, can you explain this part a little bit more?

JSON files were a one-liners for arounds two now (#12445), there is also an option to disable the minifaction during the development, and the minifcation has been added after the suggestion from Cedric (if I remember correctly.)

Reply:
Also I'm not sure if JSON files should be reviewed in-depth on every occasion, at the end there is a generated content, just like `yarn.lock`. But I can also agree that GitHub diff presentation isn't the best, which makes the review process even harder, this is how I see the diffs locally:

<img width="1507" alt="Screenshot 2021-06-17 at 10 39 01" src="https://user-images.githubusercontent.com/719641/122363274-1b2e7300-cf59-11eb-932d-48e7c8390418.png">

Reply:
> Sorry I didn't get that suggestion about 0, can you explain this part a little bit more?

Using 0 as the last parameter of `JSON.stringify` makes the string a single line, if we would use `1` or `2` instead then the JSON would look like formatted JSON, with multiple lines and indents, so Git could optimize the space that subsequent deltas take. It's not very important tho, these files are not too big and we're not going to regenerate them very often, so that's fine.

> And I can also agree that it history of Expo repo is a way different story, which is also affecting the developers or community, mainly during init/checkout, this is something that I would like to tackle in the future too.

What exactly do you have in mind? I've already rewritten the history using `git filter-repo` and deleted all old orphaned objects, which reduced the size after clone from 2.7GB to less than 700MB. It's on the `future-main` branch, but we're waiting for better times to proceed with the migration 😄 Especially some assets need to be moved/removed/compressed — we could reduce the repo size even more. Imho we should also move `docs` folder to separate repo, but we still haven't decided what to do.

> Also I'm not sure if JSON files should be reviewed in-depth on every occasion

Yeah, I think we will finally add `linguist-generated` Git attribute to these files (like `build` folders).

Reply:
> Using 0 as the last parameter of `JSON.stringify` makes the string a single line, if we would use `1` or `2` instead then the JSON would look like formatted JSON, with multiple lines and indents, so Git could optimize the space that subsequent deltas take. It's not very important tho, these files are not too big and we're not going to regenerate them very often, so that's fine.

I'm already using `0` in this context to minify the file:
* https://github.com/expo/expo/blob/master/tools/src/commands/GenerateDocsAPIData.ts#L79

API generation is still in a development process, probably at the end I will also move the categorization from `APISection` to the command in tools which will change the structure of JSON files once more (full regeneration will be needed). But the long term goal is to stabilize the format and data so only changes to package API would require file regeneration.

> What exactly do you have in mind? I've already rewritten the history using `git filter-repo` and deleted all old orphaned objects, which reduced the size after clone from 2.7GB to less than 700MB. It's on the `future-main` branch, but we're waiting for better times to proceed with the migration 😄 Especially some assets need to be moved/removed/compressed — we could reduce the repo size even more. Imho we should also move `docs` folder to separate repo, but we still haven't decided what to do.

I saw your earlier work on the size reduction and I really appreciate the effort and results, it was a bit bothersome earlier. 😉 But there are also other scopes in which the Git repo can be optimize. One of my hade right now, which will improve DX will be reduction of number of branches in main repo,which helps especially, when you have defined few remotes. I have suggested this earlier on the Discord.

> Yeah, I think we will finally add `linguist-generated` Git attribute to these files (like `build` folders).

Seems like a good idea. 🙂